packages/sharing-smart-contract/contracts/ProtectedDataSharing.sol

```diff
@@ -220,8 +235,13 @@ contract ProtectedDataSharing is ERC721Burnable, ERC721Receiver, ManageOrders, A
             protectedDatas[_collectionId][uint160(_protectedData)] != address(0),
             "ProtectedData not in collection"
         );
-        registry.safeTransferFrom(address(this), msg.sender, uint256(uint160(_protectedData)));
+        protectedDataRegistry.safeTransferFrom(
+            address(this),
+            msg.sender,
+            uint256(uint160(_protectedData))
```

Comment:
is double cast `address to uint160 to unit256` necessary?

Reply:
yeah Explicit type conversion not allowed from `address` to `uint256`. We should cast to `uint160` in a first step. As it made in the POCO